src/Cli/Commands/AddTelemetryOptions.cs

```diff
@@ -18,20 +21,71 @@ namespace Cli.Commands
     [Verb("add-telemetry", isDefault: false, HelpText = "Add telemetry for Data Api builder Application", Hidden = false)]
     public class AddTelemetryOptions : Options
     {
+#if NET8_0_OR_GREATER
+        public AddTelemetryOptions(
+            string? appInsightsConnString = null,
+            CliBool? appInsightsEnabled = null,
+            string? otelEndpoint = null,
+            CliBool? otelEnabled = null,
+            string? otelHeaders = null,
+            OtlpExportProtocol? otlpExportProtocol = null,
+            string? serviceName = null,
+            string? config = null) : base(config)
+        {
+            if (appInsightsEnabled is not null)
+            {
+                AppInsightsConnString = appInsightsConnString;
+                AppInsightsEnabled = appInsightsEnabled;
+            }
+
+            if (otelEnabled is not null)
+            {
+                OpenTelemetryEndpoint = otelEndpoint;
+                OpenTelemetryHeaders = otelHeaders;
+                OpenTelemetryServiceName = serviceName;
+                OpenTelemetryEnabled = otelEnabled;
+                OpenTelemetryOtlpExportProtocol = otlpExportProtocol;
+            }
+        }
+#else
         public AddTelemetryOptions(string appInsightsConnString, CliBool appInsightsEnabled, string? config) : base(config)
         {
             AppInsightsConnString = appInsightsConnString;
             AppInsightsEnabled = appInsightsEnabled;
         }
+#endif
 
         // Connection string for the Application Insights resource to which telemetry data should be sent.
         // This option  is required and must be provided with a valid connection string.
         [Option("app-insights-conn-string", Required = true, HelpText = "Connection string for the Application Insights resource for telemetry data")]
-        public string AppInsightsConnString { get; }
+        public string? AppInsightsConnString { get; }
 
         // To specify whether Application Insights telemetry should be enabled. This flag is optional and default value is true.
         [Option("app-insights-enabled", Default = CliBool.True, Required = false, HelpText = "(Default: true) Enable/Disable Application Insights")]
-        public CliBool AppInsightsEnabled { get; }
+        public CliBool? AppInsightsEnabled { get; }
+
+#if NET8_0_OR_GREATER
+        // Connection string for the Open Telemetry resource to which telemetry data should be sent.
+        // This option  is required and must be provided with a valid connection string.
+        [Option("otel-endpoint", Required = true, HelpText = "Endpoint for Open Telemetry for telemetry data")]
```

Comment:
this should be marked as `false`. since it's possible the user only wants to setup ApplicationInsights. Similarly, AppInsightsConnectionString should also be marked as Required=false, for the same reason.